src/cpu.jakt

```diff
@@ -1258,63 +1285,49 @@ class CPU {
                 // Zero page
                 .clock += 3
                 .pc += 2
-
-                let address = .system.read_byte(address: arg_address)
-                .system.write_byte(address: address as! u16, value: .a)
+                
+                .zero_page_set(value: .a)
```

Comment:
ahh, here's the problem (or one of them). This doesn't work since we've already modified the pc. I do the same bug in one of the videos and have to fix it.

Reply:
Oh. Yeah, you are correct, that won't work. It's probably bugged in all the places I changed it then, since I just replaced the read with the helper, and PC increments are (usually) all at the start of opcodes.